sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/TimeWindowSuite.scala

```diff
@@ -0,0 +1,92 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.AnalysisException
+
+class TimeWindowSuite extends SparkFunSuite with ExpressionEvalHelper {
+
+  test("time window is unevaluable") {
+    intercept[UnsupportedOperationException] {
+      evaluate(TimeWindow(Literal(10L), "1 second", "1 second", "0 second"))
+    }
+  }
+
+  test("validation checks") {
+    assert(TimeWindow(Literal(10L), "1 second", "1 second", "0 second").validate().isEmpty,
+      "Is a valid expression. Should not have returned a validation error.")
+    assert(TimeWindow(Literal(10L), "1 second", "2 second", "0 second").validate().isDefined,
+      "Should have thrown validation error, because slide duration is greater than window.")
+    assert(TimeWindow(Literal(10L), "1 second", "1 second", "1 minute").validate().isDefined,
+      "Should have thrown validation error, because start time is greater than slide duration.")
+    assert(TimeWindow(Literal(10L), "1 second", "1 second", "1 second").validate().isDefined,
+      "Should have thrown validation error, because start time is equal to slide duration.")
+    assert(TimeWindow(Literal(10L), "-1 second", "1 second", "0 second").validate().isDefined,
+      "Should have thrown validation error, because window duration is negative.")
+    assert(TimeWindow(Literal(10L), "0 second", "0 second", "0 second").validate().isDefined,
+      "Should have thrown validation error, because window duration is zero.")
+    assert(TimeWindow(Literal(10L), "1 second", "-1 second", "0 second").validate().isDefined,
+      "Should have thrown validation error, because slide duration is negative.")
+    assert(TimeWindow(Literal(10L), "1 second", "0 second", "0 second").validate().isDefined,
+      "Should have thrown validation error, because slide duration is zero.")
+    assert(TimeWindow(Literal(10L), "1 second", "1 second", "-2 second").validate().isDefined,
+      "Should have thrown validation error, because start time is negative.")
+  }
+
+  test("invalid intervals throw exception") {
+    val validDuration = "10 second"
+    val validTime = "5 second"
+    for (invalid <- Seq(null, " ", "\n", "\t", "2 apples")) {
+      intercept[AnalysisException] {
+        TimeWindow(Literal(10L), invalid, validDuration, validTime).windowDuration
+      }
+      intercept[AnalysisException] {
+        TimeWindow(Literal(10L), validDuration, invalid, validTime).slideDuration
+      }
+      intercept[AnalysisException] {
```

Comment:
It would be good to check the message too.  Maybe these should live in `AnalysisErrorSuite`?